### PR TITLE
Added save, load, delete functionality

### PR DESCRIPTION
**Pull request by jstmn**
Known buggy behavior: the first time you save your script, it won't immediately show up in dropdowns for save/load, only after you reload the page. It's a minor inconvenience but not a huge deal

Once there are 1 or more scripts saved, this behaviour wont be present - saving or a deleting a script will be immediately reflected in the dropdowns :)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -54,11 +54,56 @@
...
+
+
+				</div>
+
+
 				</div>
 				<div class="navbar-end">
-					<div class="tag is-dark">Ctrl + ENTER to Run</div>
+
+<!-- 					<table>
```
Comment (lshamis) [inline index.html]: Please cleanup dead code

File: index.html [hunk 2]
```diff
@@ -77,6 +122,47 @@
 	</div>
 
 	<script>
+
+		function getAllCookies(){
+		  var pairs = document.cookie.split(";");
```
Comment (lshamis) [inline index.html]: `var` has been deprecated for `let` and `const`

`var` is a global variable
`let` is a local variable
`const` is a local variable that can't be reassigned

please replace all `var`
`const` is preferred, where possible

File: index.html [hunk 3]
```diff
@@ -204,7 +304,73 @@
 						bindKey: { win: 'Ctrl-Enter', mac: 'Command-Enter' },
 						exec: vm.run_code,
 					})
+
+					// Control-S -> Save file
+					vm.editor.commands.addCommand({
+						name: 'save_code',
+						bindKey: { win: 'Ctrl-S', mac: 'Ctrl-S' },
```
Comment (lshamis) [inline index.html]: I'm not a Mac person, but shouldn't this be `Command-S`?

File: index.html [hunk 4]
```diff
@@ -204,7 +304,73 @@
...
+
+					// Get script name
+					var script_name = prompt("Enter script name:")
+
+					// If user hits the cancel button
+					if (script_name == null){
+						return
+					}
+
+					if (script_name.length > 0){
```
Comment (lshamis) [inline index.html]: In general, do error handling up front, like you did with `if (script_name == null){`
```js
if (script_name.length == 0) {
  alert("Script name can't be empty");
  return;
}
```

File: index.html [hunk 5]
```diff
@@ -204,7 +304,73 @@
...
+							code: vm.editor.getValue()}
+						cookie_str = JSON.stringify(cookie)
+
+						console.log("Saving cookie:", cookie_str)
+
+						// Save cookie to browser
+						createCookie(script_name, cookie_str, 2*420)  // save cookie to browser
+
+						// Save cookie to Vue
+						this.cookies.push(cookie)
```
Comment (lshamis) [inline index.html]: use `vm.*` instead of `this.*`.
Makes it less bug prone to move in & out of lambdas.

File: index.html [hunk 6]
```diff
@@ -204,7 +304,73 @@
...
+				delete_cookie(cookie) {
+					
+					// Remove cookie from browser
+					eraseCookie(cookie.script_name)
+
+					// Remove from vue data
+					cookie_idx = this.cookies.indexOf(cookie)
+					this.cookies.splice(cookie_idx, 1);
+
+					alert("Cookie: '"+cookie.script_name+"' deleted")
```
Comment (lshamis) [inline index.html]: Use tick notation instead of adding strings

```js
alert(`Cookie: '${cookie.script_name}' deleted`)
```

File: index.html [hunk 7]
```diff
@@ -171,12 +257,24 @@
...
+				script_name: undefined, // name of the current scripts, might be nice to have down the line
+				cookies: []
 			},
 			mounted() {
 				const vm = this
 				const ace_script = document.querySelector('#ace')
 				ace_script.onload = vm.setup_code_panel
+
+				// Load cookies from browser, save to vuedata
+				all_cookie_names = getAllCookies()
```
Comment (lshamis) [inline index.html]: `const all_cookie_names`

**Review by lshamis — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (jstmn) [reply index.html]: 👍

File: index.html (hunk 2 above)
Comment (jstmn) [reply index.html]: updated

File: index.html (hunk 3 above)
Comment (jstmn) [reply index.html]: updated with fix

File: index.html (hunk 4 above)
Comment (jstmn) [reply index.html]: updated with fix

File: index.html (hunk 5 above)
Comment (jstmn) [reply index.html]: updated with fix

File: index.html (hunk 6 above)
Comment (jstmn) [reply index.html]: updated with fix

File: index.html (hunk 7 above)
Comment (jstmn) [reply index.html]: updated with fix

**Comment by jstmn** [comment]
I've updated /jeremy_save_load with all of the changes

**Pull request merged**